ast/compile.go

```diff
@@ -1067,18 +1067,18 @@ func newSchemaParser() *schemaParser {
 	}
 }
 
-func (parser *schemaParser) parseSchema(schema interface{}) (types.Type, error) {
```

Comment:
[nit] Since most callers don't need the `propertyKey`, can we do

```go
func (parser *schemaParser) parseSchema(schema interface{}) (types.Type, error) {
  return p.parseSchemaWithPropertyKey(schema, "")
}

func (parser *schemaParser) parseSchema(schema interface{}, propertyKey string) (types.Type, error) {
    // ... 
}
```
? 🤔